### ISSUE TITLE
Cannot access private property OC\Files\Storage\Wrapper\Encryption

### ISSUE THREAD

**Issue by matt-horwood-mayden** [nextcloud/server#6171]
### Steps to reproduce
1. Share a group folder with a link
2. use share link and provide password
3. try to download any file

### Expected behaviour
the file should be downloaded

### Actual behaviour
ether an empty zip file or a 500 error

### Server configuration

**Operating system**: minibain (debian on Prasberry pi3)

**Web server:**
nginx version: nginx/1.12.1

**Database:**
mysql  Ver 14.14 Distrib 5.5.57, for debian-linux-gnu (armv7l) using readline 6.3

**PHP version:**
PHP 5.6.30-0+deb8u1 (cli) (built: Apr 14 2017 16:20:58)

**Nextcloud version:** (see Nextcloud admin page)
12.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
update from 12.0.0 to 12.0.1

**Where did you install Nextcloud from:**
nextcloud.com download page

**Signing status:**
<details>
<summary>Signing status</summary>

No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - admin_notifications: 1.0.0
  - announcementcenter: 3.1.0
  - bookmarks: 0.10.1
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.3
  - checksum: 0.3.5
  - circles: 0.12.4
  - comments: 1.2.0
  - contacts: 1.5.3
  - dashboard: 4.0.5
  - dav: 1.3.0
  - deck: 0.2.1
  - encryption: 1.6.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_accesscontrol: 1.2.4
  - files_automatedtagging: 1.2.2
  - files_downloadactivity: 1.1.1
  - files_external: 1.3.0
  - files_markdown: 1.0.1
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - groupfolders: 1.1.0
  - keeweb: 0.4.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - mail: 0.6.4
  - mood: 0.3.2
  - nextcloud_announcements: 1.1
  - notes: 2.3.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - ocsms: 1.12.1
  - ownpad: 0.5.10
  - password_policy: 1.2.2
  - polls: 0.7.3
  - provisioning_api: 1.2.0
  - qownnotesapi: 17.5.0
  - quota_warning: 1.0.1
  - radio: 0.6.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - socialsharing_diaspora: 1.0.1
  - socialsharing_email: 1.0.1
  - socialsharing_facebook: 1.0.1
  - socialsharing_googleplus: 1.0.1
  - socialsharing_twitter: 1.0.1
  - spreed: 2.0.1
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - tasks: 0.9.5
  - twofactor_backupcodes: 1.1.1
  - twofactor_totp: 1.3.1
  - updatenotification: 1.2.0
  - user_external: 0.4
  - workflowengine: 1.2.0
Disabled:
  - theming
  - user_ldap
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "oc0xc6xq5cff",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.horwood.biz",
            "212.159.117.222",
            "192.168.49.200"
        ],
        "datadirectory": "\/var\/data\/horwood_owncloud",
        "overwrite.cli.url": "https:\/\/owncloud.horwood.biz",
        "dbtype": "mysql",
        "version": "12.0.1.5",
        "dbname": "horwood_nextcloud",
        "dbhost": "192.168.49.204",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "logfile": "\/var\/data\/owncloud\/owncloud.log",
        "logdateformat": "Y-m-d H:i:s",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "nextcloud",
        "mail_domain": "horwood.biz",
        "mail_smtphost": "vps.horwood.biz",
        "mail_smtpport": "25",
        "theme": "horwood_mini",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "192.168.49.203",
            "port": 6379,
            "timeout": 0
        },
        "asset-pipeline.enabled": false,
        "filesystem_check_changes": "0",
        "trashbin_retention_obligation": "auto",
        "preview_max_scale_factor": 1,
        "preview_max_x": 1024,
        "preview_max_y": 1024,
        "maintenance": false,
        "loglevel": "1",
        "htaccess.RewriteBase": "\/",
        "has_internet_connection": true,
        "data-fingerprint": "a40dea991a946a4d82e4a3702c8f1fcc",
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
        "log_rotate_size": 52428800
    }
}
</details>

**Are you using external storage, if yes which one:** Local, Google, Dropbox

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:** Opera 48.0.2679.0 (PGO) - Opera is up to date

**Operating system:** Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>

2017/08/17 15:45:08 [error] 6862#0: *269904 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Cannot access private property OC\Files\Storage\Wrapper\Encryption::$mountPoint in /var/data/websites/3rd_party/horwood_nextcloud/apps/files_accesscontrol/lib/Operation.php on line 85" while reading upstream, client: 81.138.237.78, server: owncloud.horwood.biz, request: "GET /index.php/s/N4SgOmpLuJEq0vo/download?path=%2F HTTP/1.1", upstream: "fastcgi://192.168.49.204:9000", host: "owncloud.horwood.biz"</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

Level | App | Message | Time
-- | -- | -- | --
Error | PHP | Cannot access private property OC\Files\Storage\Wrapper\Encryption::$mountPoint at /var/data/websites/3rd_party/horwood_nextcloud/apps/files_accesscontrol/lib/Operation.php#85 | 2017-08-17 15:45:08
Info | admin_audit | The shared folder with the token "N4SgOmpLuJEq0vo" by "matt" has been accessed. | 2017-08-17 15:45:08

</details>

#### Browser log
<details>
<summary>Browser log</summary>

{
  "log": {
    "version": "1.1",
    "creator": {
      "name": "Firefox",
      "version": "52.3.0"
    },
    "browser": {
      "name": "Firefox",
      "version": "52.3.0"
    },
    "pages": [
      {
        "startedDateTime": "2017-08-17T15:48:37.821+01:00",
        "id": "page_1",
        "title": "Files - Horwood Cloud",
        "pageTimings": {
          "onContentLoad": -1,
          "onLoad": -1
        }
      }
    ],
    "entries": [
      {
        "pageref": "page_1",
        "startedDateTime": "2017-08-17T15:48:37.821+01:00",
        "time": 2285,
        "request": {
          "bodySize": 0,
          "method": "GET",
          "url": "https://owncloud.horwood.biz/index.php/s/N4SgOmpLuJEq0vo/download?path=%2F",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Host",
              "value": "owncloud.horwood.biz"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0"
            },
            {
              "name": "Accept",
              "value": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
            },
            {
              "name": "Accept-Language",
              "value": "en-GB,en;q=0.5"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate, br"
            },
            {
              "name": "Cookie",
              "value": "nc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; oc0xc6xq5cff=aosb26afbcngb6rqqd9r97v1k3; oc_sessionPassphrase=ecd99Gw5fe4nYe4MgUyCqswi7oq9qHsw0ZrJavmXq8VrADD%2BKtmW5ox1Iprks8vrcO91Mn%2BQu1QSTFthHdDGLVeRcZhu1dH%2Ft0MwdR%2FZaSmgOXA2aLcPb6CA8nNC1qAm"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Upgrade-Insecure-Requests",
              "value": "1"
            }
          ],
          "cookies": [
            {
              "name": "nc_sameSiteCookielax",
              "value": "true"
            },
            {
              "name": "nc_sameSiteCookiestrict",
              "value": "true"
            },
            {
              "name": "__Host-nc_sameSiteCookielax",
              "value": "true"
            },
            {
              "name": "__Host-nc_sameSiteCookiestrict",
              "value": "true"
            },
            {
              "name": "oc0xc6xq5cff",
              "value": "aosb26afbcngb6rqqd9r97v1k3"
            },
            {
              "name": "oc_sessionPassphrase",
              "value": "ecd99Gw5fe4nYe4MgUyCqswi7oq9qHsw0ZrJavmXq8VrADD+KtmW5ox1Iprks8vrcO91Mn+Qu1QSTFthHdDGLVeRcZhu1dH/t0MwdR/ZaSmgOXA2aLcPb6CA8nNC1qAm"
            }
          ],
          "queryString": [
            {
              "name": "path",
              "value": "/"
            }
          ],
          "postData": {
            "mimeType": "",
            "params": [],
            "text": ""
          },
          "headersSize": 710
        },
        "response": {
          "status": 200,
          "statusText": "OK",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Accept-Ranges",
              "value": "bytes"
            },
            {
              "name": "Cache-Control",
              "value": "no-store, no-cache, must-revalidate, post-check=0, pre-check=0"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Content-Disposition",
              "value": "attachment; filename*=UTF-8''Pratchett%20and%20Gaiman.zip; filename=\"Pratchett%20and%20Gaiman.zip\""
            },
            {
              "name": "Content-Transfer-Encoding",
              "value": "binary"
            },
            {
              "name": "Content-Type",
              "value": "application/zip"
            },
            {
              "name": "Date",
              "value": "Thu, 17 Aug 2017 14:48:40 GMT"
            },
            {
              "name": "Expires",
              "value": "0"
            },
            {
              "name": "Last-Modified",
              "value": "Thu, 17 Aug 2017 14:48:39 GMT"
            },
            {
              "name": "Pragma",
              "value": "public"
            },
            {
              "name": "Server",
              "value": "nginx/1.12.1"
            },
            {
              "name": "Strict-Transport-Security",
              "value": "max-age=15768000; includeSubDomains; preload;"
            },
            {
              "name": "Transfer-Encoding",
              "value": "chunked"
            },
            {
              "name": "X-Content-Type-Options",
              "value": "nosniff"
            },
            {
              "name": "X-Frame-Options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "X-Permitted-Cross-Domain-Policies",
              "value": "none"
            },
            {
              "name": "X-Powered-By",
              "value": "PHP/5.6.30-0+deb8u1"
            },
            {
              "name": "X-Robots-Tag",
              "value": "none"
            },
            {
              "name": "X-XSS-Protection",
              "value": "1; mode=block"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'self'; script-src 'self' 'unsafe-eval' 'nonce-bGpRVGV2NW5xaWQ5eEJLT1VMc29UMlZ3cnE5Z1RValNQNTZLbDV6bXJscz06cm1WRVRKZ3F6R015akN2Z0tkRk1JQUVBNStrVVBTYWJkZXZwdyttVm5nZz0='; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';"
            },
            {
              "name": "x-download-options",
              "value": "noopen"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "application/zip",
            "size": 120,
            "encoding": "base64",
            "text": "UEsDBBQAAAAAABN2EUsAAAAAAAAAAAAAAAAVAAAAUHJhdGNoZXR0IGFuZCBHYWltYW4vUEsDBBQAAAAAABN2EUsAAAAAAAAAAAAAAAAnAAAAUHJhdGNoZXR0IGFuZCBHYWltYW4vR29vZCBPbWVucyBEaXNjIDEv"
          },
          "redirectURL": "",
          "headersSize": 1150,
          "bodySize": -1
        },
        "cache": {},
        "timings": {
          "blocked": 0,
          "dns": 0,
          "connect": 0,
          "send": 0,
          "wait": 2285,
          "receive": 0
        },
        "serverIPAddress": "212.159.117.222",
        "connection": "443"
      }
    ]
  }
}
</details>

**Comment by LukasReschke** [comment]
@nickvergessen Good old:

> Cannot access private property OC\Files\Storage\Wrapper\Encryption::$mountPoint at /var/data/websites/3rd_party/horwood_nextcloud/apps/files_accesscontrol/lib/Operation.php#85
--

**Comment by nickvergessen** [comment]
Yeah its fixed for 12.0.3 and waiting for your review ;)
https://github.com/nextcloud/files_accesscontrol/pull/74/files

**Comment by matt-horwood-mayden** [comment]
I can confirm this is now fixed, ta 👍
